search/ranker.cpp

```diff
@@ -329,20 +329,59 @@ void Ranker::Init(Params const & params, Geocoder::Params const & geocoderParams
   m_tentativeResults.clear();
 }
 
-Result Ranker::MakeResult(RankerResult const & r) const
+Result Ranker::MakeResult(RankerResult const & rankerResult, bool needAddress,
+                          bool needHighlighting) const
 {
-  Result res = GenerateFinalResult(r, true /* needAddress */);
-  HighlightResult(m_params.m_tokens, m_params.m_prefix, res);
-  if (ftypes::IsLocalityChecker::Instance().GetType(r.GetTypes()) == ftypes::NONE)
+  uint32_t const type = rankerResult.GetBestType(&m_params.m_preferredTypes);
+  string name = rankerResult.GetName();
+
+  string address;
+  if (needAddress)
+  {
+    ReverseGeocoder::Address addr;
+    m_reverseGeocoder.GetNearbyAddress(rankerResult.GetCenter(), addr);
+
+    // Insert exact address (street and house number) instead of empty result name.
+    if (name.empty() && addr.GetDistance() == 0)
+      name = FormatStreetAndHouse(addr);
+
+    address = rankerResult.GetRegionName(m_infoGetter, type);
+    // Format full address only for suitable results.
+    if (ftypes::IsAddressObjectChecker::Instance()(rankerResult.GetTypes()))
```

Comment:
So, there can be cases when addr is unused. That's why I proposed you to implement a lazy wrapper around addr.

Reply:
done